wallet-new/server/Cardano/Wallet/API/V1/LegacyHandlers/Transactions.hs

```diff
@@ -118,6 +122,46 @@ estimateFees Payment{..} = do
     cAccountId <- migrate pmtSource
     utxo <- V0.getMoneySourceUtxo ws (V0.AccountMoneySource cAccountId)
     outputs <- V0.coinDistrToOutputs =<< mapM migrate pmtDestinations
-    fee <- V0.rewrapTxError "Cannot compute transaction fee" $
-        eitherToThrow =<< V0.runTxCreator policy (V0.computeTxFee pendingAddrs utxo outputs)
-    single <$> migrate fee
+    efee <- V0.runTxCreator policy (V0.computeTxFee pendingAddrs utxo outputs)
+    single <$> case efee of
+        Left txError ->
+            case txError of
+                V0.NotEnoughMoney coin -> do
+                    acct <- V0.getAccount cAccountId
+                    pure $ mkLowerBound acct coin
+                V0.NotEnoughAllowedMoney coin -> do
+                    acct <- V0.getAccount cAccountId
+                    pure $ mkLowerBound acct coin
+                _ ->
+                    throwM (transactionErrorToWalletError txError)
+        Right fee ->
+            migrate (fee, Accurate)
+  where
+    mkLowerBound
+        :: V0.CAccount
+        -- ^ The account that failed to pay.
+        -> Core.Coin
+        -- ^ The amount of coins necessary to complete the transaction.
+        -> EstimatedFees
+    mkLowerBound acct amountUnder =
+        EstimatedFees (V1 (Core.mkCoin (fromIntegral feeLowerBound))) LowerBound
+      where
+        feeLowerBound =
+            totalToSpend - txnTotal
+        totalToSpend =
+            Core.getCoin amountUnder + acctAmount
+        txnTotal =
+            sum (fmap (Core.getCoin . unV1 . pdAmount) pmtDestinations)
+        acctAmount =
+            case V0.caAmount acct of
```

Comment:
I might be wrong, but if you use the `Utxo` approach as I was suggesting, you now get a proper `Coin` type, not an awful `CCoin`, and all this code goes away.